panic: nil pointer dereference in NATS replica when object store bucket does not exist

## Summary

When the NATS JetStream Object Store bucket does not exist, `litestream replicate` logs a clean initialization error but then **panics with a nil-pointer SIGSEGV** (exit code 2) instead of exiting gracefully.

## Reproduction (main @ c96c0f4, v0.5.14+1)

1. Run a NATS server with JetStream (`nats-server -js` / `docker run nats:latest -js`) — do **not** create any object store bucket.
2. `litestream replicate /path/to/test.db "nats://localhost:4222/missing-bucket"`

First the expected error:
```
ERROR ... "enforce snapshot retention: fetch ltx files: nats: failed to initialize object store:
failed to access object store bucket \"missing-bucket\" (bucket must be created beforehand): nats: bucket not found"
```
Then:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation ...]
  nats/replica_client.go:285  (c.objectStore.List on nil objectStore)
  ... LTXFiles → MaxLTXFileInfo → checkDatabaseBehindReplica
```

Reproduced twice (missing bucket; and a plain JetStream stream created with the same name, which also isn't an object store).

## Cause

`ReplicaClient.objectStore` remains nil after `Init` fails, and `LTXFiles` (`nats/replica_client.go:285`) calls `c.objectStore.List` without checking. Other code paths that re-call Init recover; `checkDatabaseBehindReplica` hits the nil directly.

## Expected

A clean, actionable error (e.g. "bucket must be created beforehand: nats object store add <name>") and a non-panic exit.

--- a/nats/replica_client.go
+++ b/nats/replica_client.go
@@ -130,12 +130,14 @@ func (c *ReplicaClient) Init(ctx context.Context) error {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
-	if c.nc != nil {
+	if c.objectStore != nil {
 		return nil
 	}
 
-	if err := c.connect(ctx); err != nil {
-		return fmt.Errorf("nats: failed to connect: %w", err)
+	if c.nc == nil {
+		if err := c.connect(ctx); err != nil {
+			return fmt.Errorf("nats: failed to connect: %w", err)
+		}
 	}
 
 	if err := c.initObjectStore(ctx); err != nil {

--- a/nats/replica_client_test.go
+++ b/nats/replica_client_test.go
@@ -1,9 +1,13 @@
 package nats
 
 import (
+	"context"
+	"errors"
 	"testing"
 	"time"
 
+	natsgo "github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
 	"github.com/superfly/ltx"
 )
 
@@ -11,6 +15,17 @@ func TestReplicaClient_Type(t *testing.T) {
 	client := NewReplicaClient()
 	if got, want := client.Type(), "nats"; got != want {
 		t.Fatalf("Type()=%s, want %s", got, want)
+	}
+}
+
+func TestReplicaClient_LTXFiles_InitError(t *testing.T) {
+	client := NewReplicaClient()
+	client.BucketName = "missing-bucket"
+	client.nc = &natsgo.Conn{}
+	client.js = &objectStoreErrorJetStream{err: jetstream.ErrBucketNotFound}
+
+	if _, err := client.LTXFiles(t.Context(), 0, 0, false); !errors.Is(err, jetstream.ErrBucketNotFound) {
+		t.Fatalf("LTXFiles() error = %v, want %v", err, jetstream.ErrBucketNotFound)
 	}
 }
 
@@ -193,6 +208,15 @@ type mockOtherError struct{}
 
 func (e *mockOtherError) Error() string {
 	return "some other error"
+}
+
+type objectStoreErrorJetStream struct {
+	jetstream.JetStream
+	err error
+}
+
+func (js *objectStoreErrorJetStream) ObjectStore(context.Context, string) (jetstream.ObjectStore, error) {
+	return nil, js.err
 }
 
 func TestReplicaClientDefaults(t *testing.T) {

--- a/store.go
+++ b/store.go
@@ -169,26 +169,43 @@ func (s *Store) Open(ctx context.Context) error {
 		return err
 	}
 
-	g, ctx := errgroup.WithContext(ctx)
-	g.SetLimit(50)
+	initGroup, initCtx := errgroup.WithContext(ctx)
+	initGroup.SetLimit(50)
 	for _, db := range s.dbs {
 		db := db
-		g.Go(func() error {
+		initGroup.Go(func() error {
 			select {
-			case <-ctx.Done():
-				return ctx.Err()
+			case <-initCtx.Done():
+				return initCtx.Err()
 			default:
 			}
 
 			if db.Replica != nil && db.Replica.Client != nil {
-				if err := db.Replica.Client.Init(ctx); err != nil {
+				if err := db.Replica.Client.Init(initCtx); err != nil {
 					return fmt.Errorf("initialize replica client for %q: %w", db.Path(), err)
 				}
+			}
+			return nil
+		})
+	}
+	if err := initGroup.Wait(); err != nil {
+		return err
+	}
+
+	openGroup, openCtx := errgroup.WithContext(ctx)
+	openGroup.SetLimit(50)
+	for _, db := range s.dbs {
+		db := db
+		openGroup.Go(func() error {
+			select {
+			case <-openCtx.Done():
+				return openCtx.Err()
+			default:
 			}
 			return db.Open()
 		})
 	}
-	if err := g.Wait(); err != nil {
+	if err := openGroup.Wait(); err != nil {
 		return err
 	}
 

--- a/store.go
+++ b/store.go
@@ -178,8 +178,14 @@ func (s *Store) Open(ctx context.Context) error {
 			case <-ctx.Done():
 				return ctx.Err()
 			default:
-				return db.Open()
 			}
+
+			if db.Replica != nil && db.Replica.Client != nil {
+				if err := db.Replica.Client.Init(ctx); err != nil {
+					return fmt.Errorf("initialize replica client for %q: %w", db.Path(), err)
+				}
+			}
+			return db.Open()
 		})
 	}
 	if err := g.Wait(); err != nil {

--- a/store_test.go
+++ b/store_test.go
@@ -31,6 +31,51 @@ func TestStore_Open_InitError(t *testing.T) {
 	}
 }
 
+func TestStore_Open_InitErrorDoesNotOpenDBs(t *testing.T) {
+	want := errors.New("init error")
+	initStarted := make(chan struct{})
+	allowFailure := make(chan struct{})
+
+	db0 := litestream.NewDB(filepath.Join(t.TempDir(), "db0"))
+	db0.MonitorInterval = 0
+	db0.Replica = litestream.NewReplicaWithClient(db0, &mock.ReplicaClient{
+		InitFunc: func(context.Context) error {
+			close(initStarted)
+			<-allowFailure
+			return nil
+		},
+	})
+	t.Cleanup(func() {
+		if db0.IsOpen() {
+			if err := db0.Close(context.Background()); err != nil {
+				t.Errorf("close db0: %v", err)
+			}
+		}
+	})
+
+	db1 := litestream.NewDB(filepath.Join(t.TempDir(), "db1"))
+	db1.MonitorInterval = 0
+	db1.Replica = litestream.NewReplicaWithClient(db1, &mock.ReplicaClient{
+		InitFunc: func(context.Context) error {
+			<-initStarted
+			close(allowFailure)
+			return want
+		},
+	})
+
+	store := litestream.NewStore([]*litestream.DB{db0, db1}, litestream.CompactionLevels{{Level: 0}})
+	store.CompactionMonitorEnabled = false
+	if err := store.Open(t.Context()); !errors.Is(err, want) {
+		t.Fatalf("Open() error = %v, want %v", err, want)
+	}
+	if db0.IsOpen() {
+		t.Fatal("db0 opened before all replica clients initialized")
+	}
+	if db1.IsOpen() {
+		t.Fatal("db1 opened before all replica clients initialized")
+	}
+}
+
 func TestStore_CompactDB(t *testing.T) {
 	t.Run("L1", func(t *testing.T) {
 		db0, sqldb0 := testingutil.MustOpenDBs(t)

--- a/store_test.go
+++ b/store_test.go
@@ -1,6 +1,7 @@
 package litestream_test
 
 import (
+	"context"
 	"errors"
 	"fmt"
 	"path/filepath"
@@ -13,7 +14,22 @@ import (
 	"github.com/benbjohnson/litestream"
 	"github.com/benbjohnson/litestream/file"
 	"github.com/benbjohnson/litestream/internal/testingutil"
+	"github.com/benbjohnson/litestream/mock"
 )
+
+func TestStore_Open_InitError(t *testing.T) {
+	want := errors.New("init error")
+	db := litestream.NewDB(filepath.Join(t.TempDir(), "db"))
+	db.Replica = litestream.NewReplicaWithClient(db, &mock.ReplicaClient{
+		InitFunc: func(context.Context) error { return want },
+	})
+
+	store := litestream.NewStore([]*litestream.DB{db}, litestream.CompactionLevels{{Level: 0}})
+	store.CompactionMonitorEnabled = false
+	if err := store.Open(t.Context()); !errors.Is(err, want) {
+		t.Fatalf("Open() error = %v, want %v", err, want)
+	}
+}
 
 func TestStore_CompactDB(t *testing.T) {
 	t.Run("L1", func(t *testing.T) {

